Samples/VirtualCamera/VirtualCameraManager_WinRT/CameraKsPropertyInquiry.cpp

```diff
@@ -0,0 +1,215 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+
+#include "pch.h"
+#include "Common.h"
+#include "CameraKsPropertyInquiry.h"
+#include "CameraKsPropertyInquiry.g.cpp"
+#include "BasicExtendedPropertyPayload.h"
+#include "BasicCustomPropertyPayload.h"
+
+
+namespace winrt::VirtualCameraManager_WinRT::implementation
+{
+    /// <summary>
+    /// Getting an extended control payload
+    /// </summary>
+    /// <param name="extendedControlKind"></param>
+    /// <param name="controller"></param>
+    /// <returns></returns>
+    winrt::VirtualCameraManager_WinRT::IExtendedPropertyPayload CameraKsPropertyInquiry::GetExtendedControl(
+        winrt::VirtualCameraManager_WinRT::ExtendedControlKind const& extendedControlKind, 
+        winrt::Windows::Media::Devices::VideoDeviceController const& controller)
+    {
+        winrt::VirtualCameraManager_WinRT::IExtendedPropertyPayload resultPayload = nullptr;
+
+        UINT controlId = 0;
+        switch (extendedControlKind)
+        {
+        case winrt::VirtualCameraManager_WinRT::ExtendedControlKind::EyeGazeCorrection:
+            controlId = static_cast<int>(extendedControlKind);
+            break;
+        default:
+            throw winrt::hresult_invalid_argument(L"Attempting to get an extendewd control unhandled in this sample");
+        }
+
+        KsProperty prop(
+            { 0x1CB79112, 0xC0D2, 0x4213, {0x9C, 0xA6, 0xCD, 0x4F, 0xDB, 0x92, 0x79, 0x72} } /* KSPROPERTYSETID_ExtendedCameraControl */,
+            controlId /* KSPROPERTY_CAMERACONTROL_EXTENDED_* */,
+            0x00000001 /* KSPROPERTY_TYPE_GET */);
+        uint8_t* pProp = reinterpret_cast<uint8_t*>(&prop);
+
+        array_view<uint8_t const> serializedProp = array_view<uint8_t const>(pProp, sizeof(KsProperty));
+
+        // get the control payload
+        auto getResult = controller.GetDevicePropertyByExtendedId(serializedProp, nullptr);
+
+        // if the control is not supported, return null
+        if (getResult.Status() == Windows::Media::Devices::VideoDeviceControllerGetDevicePropertyStatus::NotSupported)
+        {
+            return resultPayload;
+        }
+
+        // if getting the control fails, throw
+        if (getResult.Status() != Windows::Media::Devices::VideoDeviceControllerGetDevicePropertyStatus::Success)
+        {
+            throw hresult_invalid_argument(L"Could not retrieve extended device control " + winrt::to_hstring(controlId) + L" with status: " + winrt::to_hstring((int)getResult.Status()));
```

Comment:
suggestion: use VideoDeviceControllerGetDevicePropertyStatus instead of status so it's easier to match status enum